redis/redis.d.ts

```diff
@@ -34,6 +34,10 @@ declare module "redis" {
         versions: number[];
     }
 
+    interface RetryStrategy {
+      (options: {error: Error, total_retry_time: number, times_connected: number, attempt: number}): number;
```

Comment:
I'd suggest splitting the options param into a separate interface (e.g. RetryStrategyOptions) for clarity.

Also, the function can potentially return a non-numeric value (e.g. an Error to flush all commands with an individual error, or any other non-numeric value to flush the built in error) ([Documentation here](https://github.com/NodeRedis/node_redis#options-object-properties)). Maybe the return value, at least for now, be `number | Error`? It doesn't cover all scenarios, but it will at least cover most while keeping a reasonable level of type checking.